src/modules/adaptors/Aura/AuraERC4626Adaptor.sol

```diff
@@ -0,0 +1,79 @@
+    //================= Configuration Data Specification =================
+    // NA
+    //====================================================================
+
+    /**
+     * @notice Attempted to interact with an auraPool the Cellar is not using.
+     */
+    error AuraExtrasAdaptor__AuraPoolPositionsMustBeTracked(address auraPool);
+
+    //============================================ Global Functions ===========================================
+    /**
+     * @dev Identifier unique to this adaptor for a shared registry.
+     * Normally the identifier would just be the address of this contract, but this
+     * Identifier is needed during Cellar Delegate Call Operations, so getting the address
+     * of the adaptor is more difficult.
+     */
+    function identifier() public pure virtual override returns (bytes32) {
+        return keccak256(abi.encode("Aura ERC4626 Adaptor V 0.1"));
+    }
+
+    //============================================ Strategist Functions ===========================================
+
+    /**
+     * @notice Allows strategists to get rewards for an AuraPool.
+     * @param _auraPool the specified AuraPool
+     * @param _claimExtras Whether or not to claim extra rewards associated to the AuraPool (outside of rewardToken for AuraPool)
+     */
+    function getRewards(IBaseRewardPool _auraPool, bool _claimExtras) public {
+        _validateAuraPool(address(_auraPool));
+        _getRewards(_auraPool, _claimExtras);
+    }
+
+    /**
+     * @notice Validates that a given auraPool is set up as a position in the calling Cellar.
+     * @dev This function uses `address(this)` as the address of the calling Cellar.
+     */
+    function _validateAuraPool(address _auraPool) internal view {
+        bytes32 positionHash = keccak256(abi.encode(identifier(), false, abi.encode(_auraPool)));
+        uint32 positionId = Cellar(address(this)).registry().getPositionHashToPositionId(positionHash);
+        if (!Cellar(address(this)).isPositionUsed(positionId))
+            revert AuraExtrasAdaptor__AuraPoolPositionsMustBeTracked(_auraPool); // TODO: troubleshoot uncommented implementation code here
+    }
+
+    //============================================ Interface Helper Functions ===========================================
+
+    //============================== Interface Details ==============================
+    // It is unlikely, but AURA pool interfaces can change between versions.
+    // To account for this, internal functions will be used in case it is needed to
+    // implement new functionality.
+    //===============================================================================
+
+    function _getRewards(IBaseRewardPool _auraPool, bool _claimExtras) internal virtual {
+        _auraPool.getReward(address(this), _claimExtras); // TODO: confirm that any and all reward tokens associated to this position will be transferred from this external call.
```

Comment:
Was this TODO checked?

Reply:
Done. Double checked it via `console.log` the balances for `BAL`, `AURA`, and `BPTs`, where the first two asset balances should have increased, and the latter should have stayed the same, within the test file. Below are the results. 

The green percentages are the actual normalized (USD) rewards from the test for 100e18 BPT being deposited into Aura. The reddish highlighted percentages are the BAL/AURA split according to the AURA UI on their vAPR. 

There is small discrepancy, but that may be due to other fees I'm missing here. wdyt? I could send this question to Aura too.

<img width="744" alt="image" src="https://github.com/PeggyJV/cellar-contracts/assets/131093442/d121f9ff-830c-4f71-b3a3-8037af125f8e">

Also, please note that I did not check for obscure assets that may be part of third party liquidity mining outside of $BAL and $AURA rewards. We discussed how if this strategist function did not claim specific rewards that we could deploy standalone adaptors to do so, and also handle the rewards accordingly (simply claiming them or otherwise).